pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -238,16 +242,8 @@ func (r *ReconcileMachineHealthCheck) Reconcile(ctx context.Context, request rec
 		klog.Errorf("Reconciling %s: error patching status: %v", request.String(), err)
 		return reconcile.Result{}, err
 	}
-
-	// remediate
-	for _, t := range needRemediationTargets {
-		klog.V(3).Infof("Reconciling %s: meet unhealthy criteria, triggers remediation", t.string())
-		if err := t.remediate(r); err != nil {
-			klog.Errorf("Reconciling %s: error remediating: %v", t.string(), err)
-			errList = append(errList, err)
-		}
-	}
-
+	errList = r.remediate(ctx, needRemediationTargets, errList, mhc)
+	r.cleanEMR(ctx, currentHealthy, mhc)
```

Comment:
what is EMR, and why do we need to clean it here? I think it deserves a comment :)